src/pages/AddDoctor/AddDoctor.jsx

```diff
@@ -1,5 +1,193 @@
-import React from 'react';
+import React, { useState } from 'react';
+import { v4 as uuidv4 } from 'uuid';
+import { useNavigate } from 'react-router-dom';
+import Input from '../../components/Input/Input';
 
-const AddDoctor = () => <div>AddDoctor under construction</div>;
+const defaultInput = {
+  name: '',
+  specialty: '',
+  dob: '',
+  bio: '',
+  experience: '',
+  profilePic: '',
+  consultation: '',
+};
+
+const AddDoctor = () => {
+  const [inputs, setInputs] = useState(defaultInput);
+  const [errors, setErrors] = useState([]);
+  const [success, setSuccess] = useState(null);
+  const navigate = useNavigate();
+
+  const {
+    name, specialty, bio, dob, experience, profilePic, consultation,
+  } = inputs;
+
+  const handleChange = (e) => {
+    setInputs({ ...inputs, [e.target.name]: e.target.value });
+  };
+
+  const resetFields = () => {
+    setInputs(defaultInput);
+  };
+
+  const processErrorResponse = (errorRes) => {
+    const errors = [];
+    let newError = '';
+    Object.entries(errorRes.error).forEach((error) => {
+      newError = `${error[0]} ${error[1][0]}`;
+      errors.push(newError);
+    });
+    return errors;
+  };
+
+  const setSuccessOrErrorStates = (res, data) => {
+    if (res.status === 201) {
+      setSuccess('Doctor created successfully!');
+      setErrors([]);
+      resetFields();
+      setTimeout(() => navigate('/'), 1000);
+    } else {
+      setErrors(processErrorResponse(data));
+      setSuccess(null);
+    }
+  };
+
+  const addADoctor = async () => {
+    try {
+      const res = await fetch('http://127.0.0.1:3000/api/v1/doctors', {
+        method: 'POST',
+        headers: {
+          Accept: 'application/json',
+          'Content-Type': 'application/json',
+        },
+        body: JSON.stringify({
+          name,
+          bio,
+          experience,
+          specialty,
+          profile_pic: profilePic,
+          date_of_birth: dob,
+          consultation_fee: consultation,
+        }),
+      });
+
+      const data = await res.json();
+      setSuccessOrErrorStates(res, data);
+    } catch (error) {
+      setErrors(error);
+    }
+  };
```

Comment:
Nice implementation of an async function to post a new record in the API, for the next one I'll suggest you use the redux store for the several advantages of it!

Reply:
@AlexRS90 thank you for your review 🙏 I didn't use redux for that because I thought it would be overkill at this point.